integration/sandbox_clean_remove_windows_test.go

```diff
@@ -0,0 +1,149 @@
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package integration
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1"
+)
+
+func TestSandboxRemoveWithoutIPLeakage(t *testing.T) {
+	t.Logf("Make sure host-local ipam is in use")
+	config, err := CRIConfig()
+	require.NoError(t, err)
+	fs, err := os.ReadDir(config.NetworkPluginConfDir)
+	require.NoError(t, err)
+	require.NotEmpty(t, fs)
+	f := filepath.Join(config.NetworkPluginConfDir, fs[0].Name())
+	cniConfig, err := os.ReadFile(f)
+	require.NoError(t, err)
+	if !strings.Contains(string(cniConfig), "azure-vnet-ipam") {
+		t.Skip("azure-vnet ipam is not in use")
+	}
+
+	t.Logf("Create a sandbox")
+	sbConfig := PodSandboxConfig("sandbox", "remove-without-ip-leakage")
+	sb, err := runtimeService.RunPodSandbox(sbConfig, *runtimeHandler)
+	require.NoError(t, err)
+	defer func() {
+		// Make sure the sandbox is cleaned up in any case.
+		runtimeService.StopPodSandbox(sb)
+		runtimeService.RemovePodSandbox(sb)
+	}()
+
+	t.Logf("Get pod information")
+	status, info, err := SandboxInfo(sb)
+	require.NoError(t, err)
+	ip := status.GetNetwork().GetIp()
+	require.NotEmpty(t, ip)
+	require.NotNil(t, info.RuntimeSpec.Windows)
+	netNS := info.RuntimeSpec.Windows.Network.NetworkNamespace
+	require.NotEmpty(t, netNS, "network namespace should be set")
+
+	t.Logf("Should be able to find the pod ip in host-local checkpoint")
+	checkIP := func(ip string) bool {
+		f, err := os.Open("azure-vnet-ipam.json")
+		require.NoError(t, err)
+		defer f.Close()
+
+		data, err := ioutil.ReadAll(f)
+		require.NoError(t, err)
+
+		var jsonData map[string]interface{}
+		err = json.Unmarshal(data, &jsonData)
+		require.NoError(t, err)
+
+		walkJSON := func(initial map[string]interface{}, elementNames ...string) map[string]interface{} {
+			element := initial
+			for _, name := range elementNames {
+				element = element[name].(map[string]interface{})
+			}
+
+			return element
+		}
+
+		pools := walkJSON(jsonData, "IPAM", "AddressSpaces", "local", "Pools")
+
+		ipAddr := net.ParseIP(ip)
+		var ipPool map[string]interface{}
+		for poolID, pool := range pools {
+			// Each pool will contain its key as its subnet.
+			_, ipnet, _ := net.ParseCIDR(poolID)
+			if ipnet.Contains(ipAddr) {
+				ipPool = pool.(map[string]interface{})
+				break
+			}
+		}
+
+		// Search in the IP Pool and see if it's in use or not.
+		for address, details := range walkJSON(ipPool, "Addresses") {
+			if address == ip {
+				d := details.(map[string]interface{})
+				return d["InUse"].(bool)
+			}
+		}
+
+		return false
+	}
+	require.True(t, checkIP(ip))
+
+	t.Logf("Kill sandbox container")
+	require.NoError(t, KillPid(int(info.Pid)))
```

Comment:
This will only work for process containers but I guess thats fine given the support statement at the moment